sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,485 @@
+        ((h/4 - lB/2)*J1_omega(t) + (h/4 + lB/2)*J1_omega(t))*rod_frame.x
+        >>> rod.frame.ang_vel_in(C.frame)
+        J1_omega(t)*C_frame.y
+        >>> plate.frame.ang_vel_in(C.frame)
+        J2_omega(t)*rod_frame.z + J1_omega(t)*C_frame.y
+        >>> rod.frame.dcm(C.frame)
+        Matrix([
+        [cos(J1_theta(t)), 0, -sin(J1_theta(t))],
+        [               0, 1,                 0],
+        [sin(J1_theta(t)), 0,  cos(J1_theta(t))]])
+        >>> plate.frame.dcm(C.frame)
+        Matrix([
+        [ cos(J1_theta(t))*cos(J2_theta(t)), sin(J2_theta(t)), -sin(J1_theta(t))*cos(J2_theta(t))],
+        [-sin(J2_theta(t))*cos(J1_theta(t)), cos(J2_theta(t)),  sin(J1_theta(t))*sin(J2_theta(t))],
+        [                  sin(J1_theta(t)),                0,                   cos(J1_theta(t))]])
+
+    Notes
+    ======
+
+    All symbols and dynamicsymbols used with `Joints` should be used with correct
+    assumption, i.e. positive=True else it may give incorrect results in some cases.
+
+    """
+
+    def __init__(self, name, parent, child, coordinates=None, speeds=None, parent_joint_pos=None,
+        child_joint_pos=None, parent_axis=None, child_axis=None):
+
+        super().__init__(name, parent, child, coordinates, speeds, parent_joint_pos, child_joint_pos,
+            parent_axis, child_axis)
+
+    def _generate_coordinates(self, coordinate):
+        coordinates = []
+        if coordinate is None:
+            theta = dynamicsymbols(self._name + '_theta', positive=True)
+            coordinate = theta
+        coordinates.append(coordinate)
+        return coordinates
+
+    def _generate_speeds(self, speed):
+        speeds = []
+        if speed is None:
+            omega = dynamicsymbols(self._name + '_omega', positive=True)
+            speed = omega
+        speeds.append(speed)
+        return speeds
+
+    def _orient_frames(self):
+        self._child.frame.orient_axis(self._parent.frame, self._parent_axis, 0)
+        angle, axis = self._align_axis(self._parent_axis, self._child_axis)
+        if axis != Vector(0):
+            I = ReferenceFrame('I')
+            I.orient_axis(self._child.frame, self._child_axis, 0)
+            I.orient_axis(self._parent.frame, axis, angle)
+            self._child.frame.orient_axis(I, self._parent_axis, self._coordinates[0])
+        else:
+            self._child.frame.orient_axis(self._parent.frame, self._parent_axis, self._coordinates[0])
+
+    def _set_angular_velocity(self):
+        self._child.frame.set_ang_vel(self._parent.frame, self._speeds[0] * self._parent_axis.normalize())
+
+    def _set_linear_velocity(self):
+        self._parent_joint.set_vel(self._parent.frame, 0)
+        self._child_joint.set_vel(self._parent.frame, 0)
+        self._child_joint.set_pos(self._parent_joint, 0)
+        self._child.masscenter.v2pt_theory(self._parent.masscenter, self._parent.frame, self._child.frame)
+
+
+class SlidingJoint(Joint):
```

Comment:
Maybe we should simplify call this `PrismaticJoint`, as the word "sliding" is pretty ambiguous. The planar joint is also "sliding".

Reply:
The PinJoint is also probably better named "RevoluteJoint".

https://en.wikipedia.org/wiki/Prismatic_joint

https://en.wikipedia.org/wiki/Revolute_joint

Reply:
Okay, I'll change the name.

Reply:
@moorepants `PinJoint` seems nice, shall I change it too?

Reply:
Either way is fine for pin joint. Pin is shorter.

Prismatic could be "SliderJoint" to maybe match the accepted terminology better. I'm also fine with either too. It's not too big a deal.

Reply:
`PinJoint` seems fine to me, I have changed `SlidingJoint` to `PrismaticJoint`